### DISCUSSION TITLE
Does bibtex-actions support opening pdfs based on the bibtex `file` field?

### DISCUSSION THREAD

**Discussion by teddygroves** [emacs-citar/citar#258] [Q&A] [answered]
First of all, thanks for the great package. I have jumped into the vertico ecosystem mainly thanks to having a viable ivy-bibtex replacement and am generally having a great time. I can do nearly everything I want much more conveniently.

However I have been having trouble opening pdf files based on the `file` field in my bibtex entries and would like to know if this is currently supported or planned to be in the future.

To illustrate what I can't do, I have this entry in my bibliography file:

```bibtex
@book{albertyThermodynamicsBiochemicalReactions2003,
  title = {Thermodynamics of Biochemical Reactions},
  author = {Alberty, Robert A.},
  year = {2003},
  publisher = {{Wiley-Interscience}},
  address = {{Hoboken, N.J}},
  isbn = {978-0-471-22851-6},
  language = {en},
  lccn = {QP517.T48 A42 2003},
  keywords = {Bioenergetics,Physical biochemistry,Thermodynamics},
  file = {/Users/tedgro/Reading/Zotero/storage/LNAYX7S2/Alberty - 2003 - Thermodynamics of biochemical reactions.pdf}
}
```

When I run 

```elisp
(bibtex-actions-file-open-external
 (bibtex-actions-get-value
  "file"
  (bibtex-actions-get-entry "albertyThermodynamicsBiochemicalReactions2003")))

```

in the scratch buffer the pdf opens. However running either of these

```elisp
(bibtex-actions-open '("albertyThermodynamicsBiochemicalReactions2003"))
(bibtex-actions-open-library-files '("albertyThermodynamicsBiochemicalReactions2003"))
```

produces the following message

```
"No file(s) found for (albertyThermodynamicsBiochemicalReactions2003)"
```

From a quick look at the code it seems like these both of these functions are designed around a workflow where the file  names match the bibtex keys, which as you can see from the example above isn't how my files are organised (I think I just used the default settings for Zotero/better bibtex on macos for this).

I'd like to know if I have missed the right way to open pdfs based on the `file` field, or if this isn't currently supported whether it is planned in future.

For reference here is the relevant part of my `init.el` file:

```elisp
(use-package bibtex-actions
  :ensure t
  :demand t
  :straight (:type git :host github :repo "bdarcus/bibtex-actions")
  :bind (("C-c b" . bibtex-actions-insert-citation)
         :map minibuffer-local-map
         ("M-b" . bibtex-actions-insert-preset))
  :after (embark parsebib bibtex-completion)
  :config
  (add-to-list 'embark-target-finders 'bibtex-actions-citation-key-at-point)
  (add-to-list 'embark-keymap-alist '(bib-reference . bibtex-actions-map))
  (add-to-list 'embark-keymap-alist '(citation-key . bibtex-actions-buffer-map))
  (advice-add #'completing-read-multiple
              :override #'consult-completing-read-multiple)
  (setq bibtex-actions-file-variable "file"
        bibtex-actions-bibliography '("/Users/tedgro/Reading/bibliography.bib")))

```

**Comment by bdarcus** [comment, marked as answer]
Hey - I added this as a feature request here.

https://github.com/bdarcus/bibtex-actions/issues/259

So the answer is we don't support now, but easy enough to add.

**Reply by teddygroves** [reply]
Thanks for answering so quickly!

**Reply by maasdorp** [reply]
I also want this feature to be added!

**Reply by bdarcus** [reply]
New answer: yes, as of a couple of months ago ;-)

**Reply by maasdorp** [reply]
LOL thanks! After posting, I followed the numbers and saw it got added with #260 -- now I just need to get it to work my side! ;-)
